src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+        (mainGearRotationValue * secondaryGearRotationRatioTermSimplfied
+            - secondaryGearRotationValue * mainGearRotationRatioTermSimplified) //this returns a decimal where the multiplier is different for every gear ratio
+            , mainGearRotationRatioTermSimplified) //mod it with the main gear rotation ratio term so it is within the correct range
+            + mainGearRotationRatioTermSimplified)); //for more information, see the comment in the kFactor() function
+  }
+
+  /**
+   * 
+   * @return Returns true if the gear ratio works with the neccessary number of
+   *         max rotations, false otherwise
+   */
+  public boolean validateGearRatioWithMaxRotations() {
+    boolean works = true;
+    for (int i = 0; i < maxRotations * 6; i++) {
+      double elevatorPos = i * mainGearCircumference / 6; //set elevator positon (expected position that it will check with) through sixths of a rotation until max rotations is reached
+      double epoc = CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1) > 0.9 ? 0.0
+          : CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1) < 0.1 ? 0.0
+              : CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1); //calculate what the main gear encoder value would be at each position - the ternary operators account for annoying float division rounding problems on either end (0 and 1)
+      if (Math.abs(this.getAbsolutePositionOfMainObject(epoc, //the calculated main gear encoder value
+          CRTGearRatioUtil.mod(
+              elevatorPos / mainGearCircumference * secondaryGearRotationRatioTerm /
+                  mainGearRotationRatioTerm, //the calculated secondary gear encoder value
+              1))
+          - elevatorPos) > 0.01) { //check if the difference in what the function returns is greater than a small error factor, if it is, set working to false - this is a result more annoying float division problems, but will also not work if the gear ratio doesn't work
+        works = false;
+      }
+    }
+    return works;
+  }
+
+  /**
+   * 
+   * @param mainGearRotationValue
+   * @param secondaryGearRotationValue
+   * @return Applies CRT to find the absolute position of the object attached to
+   *         the main gear
+   */
+  public double getAbsolutePositionOfMainObject(double mainGearRotationValue, double secondaryGearRotationValue) {
+    return mainGearRotationValue * mainGearCircumference
+        + getMainGearFullRotations(mainGearRotationValue, secondaryGearRotationValue) * mainGearCircumference; //multiplies the number of full rotations plus the fractional amount of rotation from the main gear encoder by the circumference to determine distance traveled by the object attached to the main gear
+  }
+
+  /**
+   * @return Returns the absolute position of the main object after one wrap around
+   */
+  public double wrapAroundAbsolutePosition(){
+    return mainGearCircumference*maxRotations;
```

Comment:
Here max rotations is used as the actual max that the gear ratio setup allows

Reply:
NOTE TO SELF: FIX THIS LATER

Reply:
See the comment I added